pin-project-internal/src/lib.rs

```diff
@@ -31,6 +31,32 @@ use syn::parse::Nothing;
 /// the field.
 /// - For the other fields, makes the unpinned reference to the field.
 ///
+/// The following methods are implemented on the original `#[pin_project]` type:
+///
+/// ```ignore
+/// fn project(&mut Pin<&mut Self>) -> ProjectedType;
+/// fn project_into(Pin<&mut Self>) -> ProjectedType;
```

Comment:
I think `project` should take `Pin<&mut Self>`, the double ptr-indirection looks very strange. The argument for why it has been added is incorrect I think, see what I wrote at https://github.com/taiki-e/pin-project/pull/47#issuecomment-530467225.